tests/beta/tapes/test_qnode.py

```diff
@@ -0,0 +1,429 @@
+        assert qn.qtape is not old_tape
+
+    def test_jacobian(self, mocker, tol):
+        """Test the jacobian computation"""
+        spy = mocker.spy(QuantumTape, "numeric_pd")
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return probs(wires=0), probs(wires=1)
+
+        qn = QNode(func, dev, h=1e-8, order=2)
+        assert qn.diff_options["h"] == 1e-8
+        assert qn.diff_options["order"] == 2
+
+        x = 0.12
+        y = 0.54
+
+        res = qn(x, y)
+        jac = qn.qtape.jacobian(dev, params=[0.45, 0.1])
+
+        assert jac.shape == (4, 2)
+
+    def test_returning_non_measurements(self):
+        """Test that an exception is raised if a non-measurement
+        is returned from the QNode."""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return 5
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            QuantumFunctionError, match="must return either a single measured observable"
+        ):
+            qn(5, 1)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return expval(qml.PauliZ(0)), 5
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            QuantumFunctionError, match="must return either a single measured observable"
+        ):
+            qn(5, 1)
+
+    def test_inconsistent_measurement_order(self):
+        """Test that an exception is raised if measurements are returned in an
+        order different to how they were queued on the tape"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            m = expval(qml.PauliZ(0))
+            return expval(qml.PauliX(1)), m
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            QuantumFunctionError,
+            match="measurements must be returned in the order they are measured",
+        ):
+            qn(5, 1)
+
+        # mutating the order should work fine
+        def func(x, y):
+            global op1, op2, op3, m1, m2
+            op1 = qml.RX(x, wires=0)
+            op2 = qml.RY(y, wires=1)
+            op3 = qml.CNOT(wires=[0, 1])
+            m1 = expval(qml.PauliZ(0))
+            m2 = expval(qml.PauliX(1))
+            return [m1, m2]
+
+        qn = QNode(func, dev)
+        qn(5, 1)  # evaluate the QNode
+        assert qn.qtape.operations == [op1, op2, op3]
+        assert qn.qtape.measurements == [m1, m2]
+
+
+class TestTFInterface:
+    """Unittests for applying the tensorflow interface"""
+
+    tf = pytest.importorskip("tensorflow", minversion="2.1")

```

Comment:
I think this will cause every test in this file to be skipped

Reply:
Also, do we need to skip these tests if TF isn't installed, since it's just checking that we raise in import error?

Reply:
Won't it simply cause this class to be skipped, since it is a class attribute? The docs for `importorskip` says it can be used at the module, class, or test function level 🤔

Reply:
Maybe it's a bug? I tried it - uninstalled TF and run `pytest tests/beta/tapes/test_qnode.py`, which results in 
![image](https://user-images.githubusercontent.com/49409390/93480421-6d0f7300-f8cb-11ea-8ba7-cb188d0918ec.png)

Reply:
Oops, thanks for catching this Tom! I've moved the statement inside the test function